### PR TITLE
Sta sample improvements

### PR DESCRIPTION
**Pull request by krish2718**
* Keep trying connection in case of any failures
  * Add support for LED0 blinking when connected, this is tied to WPA
  supplicant events
  * Handle duplicate events and sanity checks
  * Update documentation

### CODE REVIEW

**Review by sachinthegreen — APPROVED** [review]

File: samples/wifi/sta/src/main.c [hunk 1]
```diff
@@ -35,6 +36,17 @@ LOG_MODULE_REGISTER(sta, CONFIG_LOG_DEFAULT_LEVEL);
...
+/* The devicetree node identifier for the "led0" alias. */
+#define LED0_NODE DT_ALIAS(led0)
+/*
+ * A build error on this line means your board is unsupported.
+ * See the sample documentation for information on how to fix this.
+ */
+static const struct gpio_dt_spec led = GPIO_DT_SPEC_GET(LED0_NODE, gpios);
+
 static struct net_mgmt_event_callback wifi_shell_mgmt_cb;
 static struct net_mgmt_event_callback net_shell_mgmt_cb;
```
Comment (D-Triveni) [inline samples/wifi/sta/src/main.c:52]: Do we still need this semaphore?

**Review by D-Triveni — APPROVED** [review]

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/sta/README.rst [hunk 2]
```diff
@@ -20,6 +20,10 @@ Overview
 ********
 
 This sample can perform Wi-Fi operations such as connect and disconnect in the 2.4GHz and 5GHz bands depending on the capabilities of an access point.
+The sample also adds LED support to map with connection and disconnection events:
+
+    * **LED1** starts blinking when the sample is connected to the access point
+    * **LED1** stops blinking when the sample is disconnected from the access point
```
Comment (richabp) [inline samples/wifi/sta/README.rst:26]:
```suggestion
    * **LED1** starts blinking when the sample is connected to the access point.
    * **LED1** stops blinking when the sample is disconnected from the access point.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: samples/wifi/sta/prj.conf [hunk 3]
```diff
@@ -77,3 +77,7 @@ CONFIG_LOG_BUFFER_SIZE=2048
 CONFIG_POSIX_CLOCK=y
 
 CONFIG_NET_MGMT_EVENT_STACK_SIZE=2048
+
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.0.99"
```
Comment (bama-nordic) [inline samples/wifi/sta/prj.conf:81]: Please explain static v/s DHCP IP addressing behaviour for users

**Review by bama-nordic — CHANGES_REQUESTED** [review]

**Review by bama-nordic — APPROVED** [review]

**Review by sr1dh48r — APPROVED** [review]

File: samples/wifi/sta/README.rst [hunk 4]
```diff
@@ -20,9 +20,27 @@ Overview
...
+    * **LED1** starts blinking when the sample is connected to the access point.
+    * **LED1** stops blinking when the sample is disconnected from the access point.
 
 Using this sample, the development kit can connect to the specified access point in :abbr:`STA (Station)` mode.
 
+IP addressing
+=============
+The sample uses DHCP to obtain an IP address for the Wi-Fi interface, but it starts with a default static IP address
+to handle networks without DHCP servers, or if the DHCP server is not available. Successful DHCP handshake will override
+the default static IP configuration.
```
Comment (richabp) [inline samples/wifi/sta/README.rst:34]:
```suggestion
The sample uses DHCP to obtain an IP address for the Wi-Fi interface.
It starts with a default static IP address to handle networks without DHCP servers, or if the DHCP server is not available.
Successful DHCP handshake will override the default static IP configuration.
```

File: samples/wifi/sta/README.rst [hunk 5]
```diff
@@ -20,9 +20,27 @@ Overview
...
+
+The default static configuration is below and can be changed in the ``prj.conf`` file:
```
Comment (richabp) [inline samples/wifi/sta/README.rst:36]:
```suggestion
The following static configuration is by default and can be changed in the :file:`prj.conf` file:
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by richabp — APPROVED** [review]

**Comment by nordicjm** [comment]
@krish2718 needs rebase

**Pull request merged**